mobile/src/main/java/org/openhab/habdroid/background/PeriodicItemUpdateWorker.kt

```diff
@@ -22,13 +23,23 @@ import org.openhab.habdroid.ui.homescreenwidget.ItemUpdateWidget
 
 class PeriodicItemUpdateWorker(val context: Context, params: WorkerParameters) : Worker(context, params) {
     override fun doWork(): Result {
-        BackgroundTasksManager.triggerPeriodicWork(context)
-        ItemUpdateWidget.updateAllWidgets(context)
-        if (CloudMessagingHelper.needsPollingForNotifications(context)) {
-            runBlocking {
-                CloudMessagingHelper.pollForNotifications(context)
+        Log.d(TAG, "doWork()")
+        doPeriodicWork(context)
+        return Result.success()
+    }
+
+    companion object {
+        private val TAG = PeriodicItemUpdateWorker::class.java.simpleName
+
+        fun doPeriodicWork(context: Context) {
+            Log.d(TAG, "doPeriodicWork()")
+            BackgroundTasksManager.scheduleUpdatesForAllKeys(context)
+            ItemUpdateWidget.updateAllWidgets(context)
+            if (CloudMessagingHelper.needsPollingForNotifications(context)) {
+                runBlocking {
```

Comment:
This needs to happen in a background thread, but not all callers of this run in a background thread.

Reply:
Good catch 👍